src/arkode/arkode_lsrkstep.c

```diff
@@ -0,0 +1,1046 @@
+    Xvecs[3] = ark_mem->tempv2;
+
+    retval = N_VLinearCombination(4, cvals, Xvecs, ark_mem->tempv1);
+    if (retval != 0) { return (ARK_VECTOROP_ERR); }
+
+    *dsmPtr = N_VWrmsNorm(ark_mem->tempv1, ark_mem->ewt);
+    lsrkStep_SprRadUpdateLogic(ark_mem, step_mem, *dsmPtr);
+  }
+  return (ARK_SUCCESS);
+}
+
+int lsrkStep_TakeStepRKG(ARKodeMem ark_mem, sunrealtype* dsmPtr, int* nflagPtr)
+{
+  printf("\nRKG is not supported yet! Try RKC or RKL instead.\n");
+  return (ARK_ILL_INPUT);
+}
+
+/*===============================================================
+  Internal utility routines
+  ===============================================================*/
+
+/*---------------------------------------------------------------
+  lsrkStep_AccessARKODEStepMem:
+
+  Shortcut routine to unpack both ark_mem and step_mem structures
+  from void* pointer.  If either is missing it returns ARK_MEM_NULL.
+  ---------------------------------------------------------------*/
+int lsrkStep_AccessARKODEStepMem(void* arkode_mem, const char* fname,
+                                 ARKodeMem* ark_mem, ARKodeLSRKStepMem* step_mem)
+{
+  /* access ARKodeMem structure */
+  if (arkode_mem == NULL)
+  {
+    arkProcessError(NULL, ARK_MEM_NULL, __LINE__, fname, __FILE__,
+                    MSG_ARK_NO_MEM);
+    return (ARK_MEM_NULL);
+  }
+  *ark_mem = (ARKodeMem)arkode_mem;
+
+  /* access ARKodeLSRKStepMem structure */
+  if ((*ark_mem)->step_mem == NULL)
+  {
+    arkProcessError(*ark_mem, ARK_MEM_NULL, __LINE__, fname, __FILE__,
+                    MSG_LSRKSTEP_NO_MEM);
+    return (ARK_MEM_NULL);
+  }
+  *step_mem = (ARKodeLSRKStepMem)(*ark_mem)->step_mem;
+  return (ARK_SUCCESS);
+}
+
+/*---------------------------------------------------------------
+  lsrkStep_AccessStepMem:
+
+  Shortcut routine to unpack the step_mem structure from
+  ark_mem.  If missing it returns ARK_MEM_NULL.
+  ---------------------------------------------------------------*/
+int lsrkStep_AccessStepMem(ARKodeMem ark_mem, const char* fname,
+                           ARKodeLSRKStepMem* step_mem)
+{
+  /* access ARKodeLSRKStepMem structure */
+  if (ark_mem->step_mem == NULL)
+  {
+    arkProcessError(ark_mem, ARK_MEM_NULL, __LINE__, fname, __FILE__,
+                    MSG_LSRKSTEP_NO_MEM);
+    return (ARK_MEM_NULL);
+  }
+  *step_mem = (ARKodeLSRKStepMem)ark_mem->step_mem;
+  return (ARK_SUCCESS);
+}
+
+/*---------------------------------------------------------------
+  lsrkStep_CheckNVector:
+
+  This routine checks if all required vector operations are
+  present.  If any of them is missing it returns SUNFALSE.
+  ---------------------------------------------------------------*/
+sunbooleantype lsrkStep_CheckNVector(N_Vector tmpl)
+{
+  if ((tmpl->ops->nvclone == NULL) || (tmpl->ops->nvdestroy == NULL) ||
+      (tmpl->ops->nvlinearsum == NULL) || (tmpl->ops->nvconst == NULL) ||
+      (tmpl->ops->nvscale == NULL) || (tmpl->ops->nvwrmsnorm == NULL))
```

Comment:
Did you verify that this is the exact set of N_Vector routines that are called by LSRKStep?